src/code/z_player_lib.c

```diff
@@ -1,153 +1,1021 @@
+        }
+    }
+}
+
+extern s16 D_801BFE14[][18];
+
+// OoT's Player_SetBootData?
+void func_80123140(GlobalContext* globalCtx, Player* player) {
+    s16* bootRegs;
+    s32 currentBoots;
+    f32 scale;
+
+    if ((player->actor.id == ACTOR_PLAYER) && (player->transformation == PLAYER_FORM_FIERCE_DEITY)) {
+        REG(27) = 0x4B0;
+    } else {
+        REG(27) = 0x7D0;
+    }
+
+    REG(48) = 0x172;
+
+    currentBoots = player->currentBoots;
+    if (currentBoots >= PLAYER_BOOTS_ZORA_UNDERWATER) {
+        if (player->stateFlags1 & PLAYER_STATE1_8000000) {
+            currentBoots++;
+        }
+        if (player->transformation == PLAYER_FORM_GORON) {
+            REG(48) = 0xC8;
+        }
+    } else if (currentBoots == PLAYER_BOOTS_GIANT) {
+        REG(48) = 0xAA;
+    }
+
+    bootRegs = D_801BFE14[currentBoots];
+    REG(19) = bootRegs[0];
+    REG(30) = bootRegs[1];
+    REG(32) = bootRegs[2];
+    REG(34) = bootRegs[3];
+    REG(35) = bootRegs[4];
+    REG(36) = bootRegs[5];
+    REG(37) = bootRegs[6];
+    REG(38) = bootRegs[7];
+    REG(39) = bootRegs[8];
+    REG(43) = bootRegs[9];
+    R_RUN_SPEED_LIMIT = bootRegs[10];
+    REG(68) = bootRegs[11];
+    REG(69) = bootRegs[12];
+    IREG(66) = bootRegs[13];
+    IREG(67) = bootRegs[14];
+    IREG(68) = bootRegs[15];
+    IREG(69) = bootRegs[16];
+    MREG(95) = bootRegs[17];
+
+    if (globalCtx->roomCtx.currRoom.unk3 == 2) {
+        R_RUN_SPEED_LIMIT = 500;
+    }
+
+    if ((player->actor.id == ACTOR_PLAYER) && (player->transformation == PLAYER_FORM_FIERCE_DEITY)) {
+        scale = 0.015f;
+    } else {
+        scale = 0.01f;
+    }
+
+    Actor_SetScale(&player->actor, scale);
+}
+
+s32 Player_InBlockingCsMode(GlobalContext* globalCtx, Player* player) {
+    return (player->stateFlags1 & (PLAYER_STATE1_20000000 | PLAYER_STATE1_200 | PLAYER_STATE1_80)) ||
+           player->csMode != 0 || globalCtx->sceneLoadFlag == 0x14 || globalCtx->unk_18B4A != 0 ||
+           (player->stateFlags1 & PLAYER_STATE1_1) || (player->stateFlags3 & PLAYER_STATE3_80) ||
+           globalCtx->actorCtx.unk268 != 0;
+}
+
+s32 Player_InCsMode(GlobalContext* globalCtx) {
+    Player* player = GET_PLAYER(globalCtx);
+
+    return Player_InBlockingCsMode(globalCtx, player) || player->unk_AA5 == 5;
+}
+
+s32 func_80123420(Player* player) {
+    return player->stateFlags3 & PLAYER_STATE3_80000000;
+}
+
+s32 func_80123434(Player* player) {
+    return player->stateFlags1 & (PLAYER_STATE1_40000000 | PLAYER_STATE1_20000 | PLAYER_STATE1_10000);
+}
+
+// Unused
+s32 func_80123448(GlobalContext* globalCtx) {
+    Player* player = GET_PLAYER(globalCtx);
+
+    return (player->stateFlags1 & PLAYER_STATE1_400000) &&
+           (player->transformation != PLAYER_FORM_HUMAN || (!func_80123434(player) && player->unk_730 == 0));
+}
+
+s32 func_801234B0(Player* player) {
+    return player->transformation == PLAYER_FORM_GORON || player->transformation == PLAYER_FORM_DEKU;
+}
```

Comment:
Well, we can probably name this, at least. And I don't really like the logical operations without brackets... we don't all have C's order of operations memorised.

Reply:
This function is very dumb imo
It is used twice on player_lib, and 9 times on player_actor.
I was hoping player_actor may give some of explanation to this dumb function, but there's probably isn't one and this is just `Player_IsGoronOrDeku`

Reply:
Yes, I can't think of any commonality between them.

Reply:
Neither can swim
(well, someone managed to spawn in pinnacle rock as deku and he does have a swimming animation, but pretty sure that was undefined behavior)

Reply:
Neither being able to swim is interesting. In that case this function could be something like `Player_FormCantSwim` or similar.
But nothing in this file makes it obvious that was the intention for this function. This is why I wanted to wait for player_actor before naming this function

Reply:
Something to keep in consideration, this function is called by `Player_ActionToModelGroup`.
The OoT version of that function looks like this:
```c

s32 Player_IsChildWithHylianShield(Player* this) {
    return gSaveContext.linkAge != LINK_AGE_ADULT && (this->currentShield == PLAYER_SHIELD_HYLIAN);
}

s32 Player_ActionToModelGroup(Player* this, s32 actionParam) {
    s32 modelGroup = sActionModelGroups[actionParam];

    if ((modelGroup == 2) && Player_IsChildWithHylianShield(this)) {
        return 1;
    } else {
        return modelGroup;
    }
}

```

The only similarity I can see here is crouching when you use the shield

Reply:
oh yeah I just wanted to point out one thing I noticed they shared, it might be irrelevant to how the function is used.

Although... naming it `Player_IsGoronOrDeku` now will help us understand what the functions that use it are doing, we can re-name it later if we find out its real purpose. Personally, I prefer naming functions before we have perfect names, as seeing the functions in later functions will help us understand what those are doing (or maybe I'm just too lazy to look up every unnamed function)

Reply:
Ah, that actually makes perfect sense, although Goron can't move.

Reply:
I'd put a comment above it rather than rename it, I think. It's useful to have the vram handy, and an IDE will show the comment on hover.

Reply:
Considering this function is only used on player_lib and player_actor, I wouldn't be opposed to name it now and rename it when we have a better understanding of it

Reply:
In which case at least add a todo comment with some hypotheses.

Reply:
Sounds good